### PR TITLE
make ATen/native/cuda/MultiLabelMarginCriterion.cu data_ptr-correct

### PR DESCRIPTION
**Pull request by dagitses**
make ATen/native/cuda/MultiLabelMarginCriterion.cu data_ptr-correct

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: aten/src/ATen/native/cuda/MultiLabelMarginCriterion.cu [hunk 1]
```diff
@@ -259,10 +259,10 @@ void multilabel_margin_loss_forward_out_cuda_template(
...
             multilabel_margin_loss_forward_kernel<scalar_t, accscalar_t>
                 <<<blocks, threads, 0, at::cuda::getCurrentCUDAStream()>>>(
-                    output_tmp.data_ptr<scalar_t>(),
-                    input_.data_ptr<scalar_t>(),
-                    target_.data_ptr<int64_t>(),
-                    is_target_.data_ptr<scalar_t>(),
+                    output_tmp.mutable_data_ptr<scalar_t>(),
+                    input_.const_data_ptr<scalar_t>(),
+                    target_.const_data_ptr<int64_t>(),
+                    is_target_.mutable_data_ptr<scalar_t>(),
```
Comment (ezyang) [inline aten/src/ATen/native/cuda/MultiLabelMarginCriterion.cu:265]: This looks wrong

**Review by ezyang — CHANGES_REQUESTED** [review]
discrepancies

File: aten/src/ATen/native/cuda/MultiLabelMarginCriterion.cu (hunk 1 above)
Comment (dagitses) [reply aten/src/ATen/native/cuda/MultiLabelMarginCriterion.cu:265]: Yes it does look wrong, but look at line 78 of this file.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**